central/auth/service/service_impl_test.go

```diff
@@ -126,3 +181,428 @@ func (s *authServiceAccessControlTestSuite) TestAuthServiceResponse() {
+			resp, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+				Config: tc.config,
+			})
+			s.NoError(err)
+			s.NotEmpty(resp.GetConfig().GetId())
+		})
+	}
+}
+
+func (s *authServiceAccessControlTestSuite) TestListConfigs() {
+	config1, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config2, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "somewhere",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "the",
+					Role:  testRole2,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	configs, err := s.svc.ListAuthMachineToMachineConfigs(s.accessCtx, nil)
+	s.NoError(err)
+
+	s.ElementsMatch(configs.GetConfigs(), []*v1.AuthMachineToMachineConfig{config1.GetConfig(), config2.GetConfig()})
+}
+
+func (s *authServiceAccessControlTestSuite) TestUpdateConfig() {
+	config, err := s.svc.AddAuthMachineToMachineConfig(s.accessCtx, &v1.AddAuthMachineToMachineConfigRequest{
+		Config: &v1.AuthMachineToMachineConfig{
+			TokenExpirationDuration: "1h",
+			Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+			Mappings: []*v1.AuthMachineToMachineConfig_Mapping{
+				{
+					Key:   "sub",
+					Value: "something",
+					Role:  testRole1,
+				},
+				{
+					Key:   "aud",
+					Value: "github",
+					Role:  testRole3,
+				},
+			},
+		},
+	})
+	s.Require().NoError(err)
+
+	config.GetConfig().Mappings = []*v1.AuthMachineToMachineConfig_Mapping{
+		{
+			Key:   "sub",
+			Value: "someone",
+			Role:  testRole2,
+		},
+	}
+
+	updatedConfig, err := s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx,
+		&v1.UpdateAuthMachineToMachineConfigRequest{Config: config.GetConfig()})
+	s.NoError(err)
+	s.Equal(config.GetConfig(), updatedConfig.GetConfig())
+
+	newConfig := &v1.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		TokenExpirationDuration: "1m",
+		Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+	}
+	updatedConfig, err = s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx, &v1.UpdateAuthMachineToMachineConfigRequest{
+		Config: newConfig,
+	})
+	s.NoError(err)
+	s.Equal(updatedConfig.GetConfig(), newConfig)
```

Comment:
(This case is more prominent than others, so I'll comment here although the similar applies to `Add` and `Remove` tests.)

What do you think of splitting tests in "logical" pieces and giving them more descriptive names?

E.g. `TestUpdateExistingConfig` and `TestUpdateAddsConfig`.

The advantage is that test methods become smaller, failures more isolated, and, most importantly, it becomes harder for future maintainers delete some tested scenario when it has an explicit name.